nixos/modules/hardware/video/nvidia.nix

```diff
@@ -214,14 +255,14 @@ in
     };
 
     services.xserver.serverLayoutSection = optionalString syncCfg.enable ''
-      Inactive "Device-modesetting[0]"
+      Inactive "Device-${igpuDriver}[0]"
     '' + optionalString offloadCfg.enable ''
       Option "AllowNVIDIAGPUScreens"
     '';
 
     services.xserver.displayManager.setupCommands = optionalString syncCfg.enable ''
       # Added by nvidia configuration module for Optimus/PRIME.
-      ${pkgs.xorg.xrandr}/bin/xrandr --setprovideroutputsource modesetting NVIDIA-0
+      ${pkgs.xorg.xrandr}/bin/xrandr --setprovideroutputsource ${igpuDriver} NVIDIA-0
```

Comment:
The configuration looks to be basically the same for Intel and AMD, so not sure why you're adding logic?

Reply:
The AMD config doesn't use the modesetting driver, and calling it that would be misleading.

Reply:
The arch wiki uses `radeon` instead of `amdgpu` in it's example, though I'm not sure what it is for AMD. This should just be a projection from different sinks from `xrandr --listproviders`, i.e., Provider 1 -> 2

```
Provider 0: id: 0x46 cap: 0xf, Source Output, Sink Output, Source Offload, Sink Offload crtcs: 3 outputs: 4 associated providers: 0 name:modesetting
Provider 1: id: 0x258 cap: 0x0 crtcs: 0 outputs: 0 associated providers: 0 name:NVIDIA-G0
```

I'm not sure about naming, if they both use `modesetting` internally then I would leave it as `modesetting` otherwise it's fine to leave it.